rs/sns/swap/src/swap.rs

```diff
@@ -910,14 +911,20 @@ impl Swap {
                         total_participant_icp_e8s,
                     );
 
+                    #[allow(deprecated)] // TODO(NNS1-3198): Remove once hotkey_principal is removed
                     let Some(parsed_principal) =
-                        string_to_principal(&cf_participant.hotkey_principal)
+                    // TODO(NNS1-3198): Simplify once hotkey_principal is removed
```

Comment:
My reasoning - these are technically two separate tasks:

1. Remove `#[allow(deprecated)]` (which IIRC does not emit a warning if it is unnecessary)
2. Simplifying the match (for which a comment isn't strictly necessary because once we remove hotkey_principal it will be obvious that the match is unnecessary, but if we decide to just stop supporting it rather than removing the field it may be helpful)